examples/mountain_wave.jl

```diff
@@ -1,72 +1,166 @@
+# # Mountain waves with open boundaries
+#
+# This example simulates mountain waves generated by flow over topography.
+# Air flowing over a hill creates internal gravity waves that propagate
+# upward and downstream. This classic atmospheric dynamics problem
+# demonstrates several features:
+#
+# - Open boundary conditions using `PerturbationAdvection`
+# - Immersed boundary grids with partial cells for topography
+# - Sponge layer to absorb waves near the top boundary
+
 using Breeze
-using Oceananigans.Grids: ImmersedBoundaryGrid, PartialCellBottom
+using Oceananigans.Grids: ImmersedBoundaryGrid, PartialCellBottom, minimum_xspacing, znode
 using Oceananigans.Units
 using Printf
+using CairoMakie
+
+# ## Domain and topography
+#
+# We set up a tall, wide domain to allow wave propagation.
+# The topography is a Gaussian hill modulated by a cosine function,
+# creating a corrugated ridge.
+
+Nx, Nz = 128, 256
+H = 20kilometers
 
-Nx, Nz = 512, 512
-H, L = 20kilometers, 200kilometers
+underlying_grid = RectilinearGrid(CPU(); size = (Nx, Nz), halo = (4, 4),
+                                  x = (-2H, 4H), z = (0, H),
+                                  topology = (Bounded, Flat, Bounded))
 
-underlying_grid = RectilinearGrid(size = (Nx, Nz), halo = (4, 4),
-                                  x = (-L, L), z = (0, H),
-                                  topology = (Periodic, Flat, Bounded))
+h₀ = 1000            # maximum hill height (m)
+a = 5kilometers      # hill width scale
+λ = 4kilometers      # corrugation wavelength
 
-h₀ = 250meters
-a = 5kilometers
-λ = 4kilometers
 hill(x) = h₀ * exp(-(x / a)^2) * cos(π * x / λ)^2
 grid = ImmersedBoundaryGrid(underlying_grid, PartialCellBottom(hill))
 
-model = AtmosphereModel(grid, advection = WENO())
+# ## Open boundary conditions
+#
+# Open boundaries allow flow to enter and exit the domain naturally.
+# The `PerturbationAdvection` scheme advects perturbations out of the
+# domain while maintaining the background flow.
+
+Uᵢ = 10  # background velocity (m/s)
+Δx = minimum_xspacing(grid)
+τ = 10 * Δx / Uᵢ  # relaxation timescale
+
+scheme = PerturbationAdvection(outflow_timescale=τ, inflow_timescale=τ)
+open_bc = OpenBoundaryCondition(Uᵢ; scheme)
+boundary_conditions = (; u = FieldBoundaryConditions(west=open_bc, east=open_bc))
```

Comment:
@navidcy i think the problem is that I am working with `u` rather than `rho * u`.